themes/default/content/blog/local-testing-with-pulumi/index.md

```diff
@@ -0,0 +1,151 @@
+        name: assetsRegistryImage.then(assetsRegistryImage => assetsRegistryImage.name),
+        pullTriggers: [assetsRegistryImage.then(assetsRegistryImage => assetsRegistryImage.sha256Digest)],
+    }, { retainOnDelete: true });
+    assetsImageRef = assetsRemoteImage.repoDigest;
+};
+```
+
+{{% notes %}}
+You can see that the released container image is hard-coded in this example. You could also pass this in as a configuration value to allow the Pulumi program to more easily account for new releases.
+{{% /notes %}}
+
+"By setting the flag for the assets service to `build` and all the other services to something else---the code only builds if the value is set to `build`---then I can isolate changes made to that service for testing, to be sure I haven't broken something elsewhere."
+
+"That seems useful."
+
+"Indeed! And you can see here that we are building a Docker network for these containers; this is what enables connectivity between the various parts of the online store when it's running locally via Docker."
+
+```typescript
+// Create a network
+const network = new docker.Network("network", {
+    name: "zephyr-net",
+});
+
+// Create an assets container
+const assetsContainer = new docker.Container("assets-container", {
+    capabilities: {
+        drops: ["ALL"],
+    },
+    envs: [
+        "PORT=8080",
+    ],
+    hostname: "assets",
+    image: assetsImageRef,
+    memory: 64,
+    name: "assets",
+    networksAdvanced: [
+        {
+            name: network.name,
+        },
+    ],
+    restart: "always",
+});
+```
+
+Bob nodded. "Yep, this all makes sense. Interesting, I hadn't really considered Pulumi for this particular use case. Usually when folks mention Pulumi the first thought in my head is provisioning cloud infrastructure."
+
+"Oh, certainly, and it works great for that---as evidenced by the fact that your team, the platform team, uses it to provision all the cloud infrastructure for Zephyr's online store, including [multiple stacks](/blog/iac-recommended-practices-structuring-pulumi-projects/) and [stack references](/blog/iac-recommended-practices-using-stack-references/) to share information across stacks."
+
+"And since we are already using Pulumi for deploying to Kubernetes in test and production, then this brings some consistency across environments. It's just a `pulumi up` regardless of whether you're running the app locally or publishing to a Kubernetes cluster in your per-developer stack, the test stack, or even the prod stack."
+
+"Exactly. Oh, look here. It's done. Looks like it took X minutes, and now our online store app is available at this URL on `localhost`."
+
+![Screenshot of a completed Pulumi operation](meta.png)
+
+Interested in seeing how you might be able to speed up your inner dev loop? Check out Alice's Pulumi program for deploying locally to Docker by navigating to the zephyr-app repository. Then select the `inner-dev-loop` tag and go to the `develop/pulumi` folder. Give it a try!
```

Comment:
There's a break in the narrative here that needs some sort of a transition -- i.e., to pull you out of the world of Alice and Bob and back into the land of "person reading this blog post."

Reply:
Agreed. I've added some text around the workflow, and added a heading to help the reader "break out" of the conversation. This is all in db276cf.